kore/src/Kore/Rewrite/RewriteStep.hs

```diff
@@ -141,14 +138,13 @@ finalizeAppliedRule
 The parts of the applied condition were already combined by 'unifyRule'. First, the @ensures@ clause is simplified under the applied condition. Then, the conditions are conjoined and simplified again.
 -}
 constructConfiguration ::
-    MonadSimplify simplifier =>
     -- | SideCondition containing metadata
     SideCondition RewritingVariableName ->
     -- | Applied condition
     Condition RewritingVariableName ->
     -- | Final configuration
     Pattern RewritingVariableName ->
-    LogicT simplifier (Pattern RewritingVariableName)
+    LogicT (LogicT Simplifier) (Pattern RewritingVariableName)
```

Comment:
Interesting that the `LogicT` is nested here. Can you try to flatten this (in a separate PR) as an experiment?